CourseGrab/Home/HomeTableViewCell.swift

```diff
@@ -8,7 +8,152 @@
+        contentView.addSubview(container)
+        
+        courseLabel.translatesAutoresizingMaskIntoConstraints = false
+        courseLabel.font = .boldSystemFont(ofSize: 14)
+        courseLabel.textColor = .courseGrabGray
+        container.addSubview(courseLabel)
+        
+        enrollButton.translatesAutoresizingMaskIntoConstraints = false
+        enrollButton.layer.cornerRadius = 2
+        enrollButton.setTitleColor(.white, for: .normal)
+        enrollButton.setTitle("ENROLL", for: .normal)
+        enrollButton.titleLabel?.font = .systemFont(ofSize: 12)
+        enrollButton.backgroundColor = .black
+        container.addSubview(enrollButton)
+        
+        removeButton.translatesAutoresizingMaskIntoConstraints = false
+        removeButton.layer.cornerRadius = 2
+        removeButton.setTitleColor(.courseGrabRuby, for: .normal)
+        removeButton.setTitle("REMOVE", for: .normal)
+        removeButton.titleLabel?.font = .systemFont(ofSize: 12)
+        removeButton.layer.borderColor = UIColor.courseGrabRuby.cgColor
+        removeButton.layer.borderWidth = 1
+        container.addSubview(removeButton)
+        
+        sectionLabel.translatesAutoresizingMaskIntoConstraints = false
+        sectionLabel.font = .boldSystemFont(ofSize: 14)
+        sectionLabel.textColor = .courseGrabGray
+        container.addSubview(sectionLabel)
+        
+        statusBadge.translatesAutoresizingMaskIntoConstraints = false
+        statusBadge.contentMode = .scaleAspectFit
+        container.addSubview(statusBadge)
+        
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.font = .systemFont(ofSize: 16)
+        titleLabel.numberOfLines = 0
+        container.addSubview(titleLabel)
+        
+        // setup constraints
+
+        container.snp.makeConstraints { make in
+            make.topMargin.equalTo(contentView).offset(6)
+            make.bottomMargin.equalTo(contentView).inset(6)
+            make.leadingMargin.equalTo(20)
+            make.trailingMargin.equalTo(contentView).inset(20)
+        }
+        
+        courseLabel.snp.makeConstraints { make in
+            make.top.equalTo(titleLabel.snp.bottom).offset(12)
+            make.trailing.equalTo(container).inset(16)
+        }
+        
+        enrollButton.snp.makeConstraints { make in
+            make.top.equalTo(sectionLabel.snp.bottom).offset(12)
+            make.leading.equalTo(container.snp.centerX).offset(9)
+            make.height.equalTo(24)
+            make.trailing.equalTo(container).inset(16)
+            make.bottom.equalTo(container).inset(16)
+        }
+        
+        removeButton.snp.makeConstraints { make in
+            make.top.equalTo(sectionLabel.snp.bottom).offset(12)
+            make.leading.equalTo(16)
+            make.height.equalTo(24)
+            make.trailing.equalTo(container.snp.centerX).inset(9)
+            make.bottom.equalTo(container).inset(16)
```

Comment:
i think theres a constraint error here? you specify both the `top & bottom` constraint as well as the `height` but you shouldn't need both

Reply:
I think I do need all of them because `container` does not have an intrinsic height, so its height is defined by the constraints of its subviews in relation to it.